common/pagination/interface.go

```diff
@@ -66,6 +66,7 @@ type (
 	Writer interface {
 		Add(Entity) error
 		Flush() error
+		FlushIfNotEmpty() error
```

Comment:
Is there any situation where we want to `Flush` when it's empty?

Reply:
You potentially can use this pagination writer to flush a page which body is empty but has no empty tags.

Reply:
Or said more generally the actual page contents might only be one input what is actually written.